pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -241,11 +242,24 @@ func (s *completed) Execute(msg dispatcher.DIDCommMsg, ctx context) (state, erro
 		if msg.Outbound {
 			return nil, fmt.Errorf("outbound responses are not allowed for state %s", s.Name())
 		}
-		// send ACK
+		response := &Response{}
+		err := json.Unmarshal(msg.Payload, response)
+		if err != nil {
+			return nil, err
```

Comment:
better to return an error with a message. same for other cases. [Ref](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/framework/context/context.go#L36).